mopidy/audio/actor.py

```diff
@@ -299,80 +275,96 @@ def on_playbin_state_changed(self, old_state, new_state, pending_state):
+        gst_logger.debug(
+            'Got ERROR bus message: error=%r debug=%r', error_msg, debug_msg)
+        gst_logger.error('GStreamer error: %s', error_msg)
         # TODO: is this needed?
         self._audio.stop_playback()
 
     def on_warning(self, error, debug):
-        gst_logger.warning(str(error).decode('utf-8'))
-        if debug:
-            gst_logger.debug(debug.decode('utf-8'))
+        error_msg = str(error).decode('utf-8')
+        debug_msg = debug.decode('utf-8')
+        gst_logger.warning('GStreamer warning: %s', error_msg)
+        gst_logger.debug(
+            'Got WARNING bus message: error=%r debug=%r', error_msg, debug_msg)
 
     def on_async_done(self):
-        gst_logger.debug('Got async-done.')
+        gst_logger.debug('Got ASYNC_DONE bus message.')
 
     def on_tag(self, taglist):
-        tags = utils.convert_taglist(taglist)
+        tags = tags_lib.convert_taglist(taglist)
+        gst_logger.debug('Got TAG bus message: tags=%r', dict(tags))
         self._audio._tags.update(tags)
         logger.debug('Audio event: tags_changed(tags=%r)', tags.keys())
         AudioListener.send('tags_changed', tags=tags.keys())
 
     def on_missing_plugin(self, msg):
-        desc = gst.pbutils.missing_plugin_message_get_description(msg)
-        debug = gst.pbutils.missing_plugin_message_get_installer_detail(msg)
-
-        gst_logger.debug('Got missing-plugin message: description:%s', desc)
+        desc = GstPbutils.missing_plugin_message_get_description(msg)
+        debug = GstPbutils.missing_plugin_message_get_installer_detail(msg)
+        gst_logger.debug(
+            'Got missing-plugin bus message: description=%r', desc)
         logger.warning('Could not find a %s to handle media.', desc)
-        if gst.pbutils.install_plugins_supported():
+        if GstPbutils.install_plugins_supported():
             logger.info('You might be able to fix this by running: '
                         'gst-installer "%s"', debug)
         # TODO: store the missing plugins installer info in a file so we can
         # can provide a 'mopidy install-missing-plugins' if the system has the
         # required helper installed?
 
-    def on_new_segment(self, update, rate, format_, start, stop, position):
-        gst_logger.debug('Got new-segment event: update=%s rate=%s format=%s '
-                         'start=%s stop=%s position=%s', update, rate,
-                         format_.value_name, start, stop, position)
-        position_ms = position // gst.MSECOND
-        logger.debug('Audio event: position_changed(position=%s)', position_ms)
-        AudioListener.send('position_changed', position=position_ms)
-
-    def on_stream_changed(self, uri):
-        gst_logger.debug('Got stream-changed message: uri=%s', uri)
-        logger.debug('Audio event: stream_changed(uri=%s)', uri)
+    def on_stream_start(self):
+        gst_logger.debug('Got STREAM_START bus message')
+        uri = self._audio._pending_uri
```

Comment:
I'm not saying this is the only way, just that I haven't gotten any smarter in this area since Dec 4 and the current version passes all our existing tests for this and seems to work in practice.